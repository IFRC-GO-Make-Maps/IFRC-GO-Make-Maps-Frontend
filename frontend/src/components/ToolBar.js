import React from 'react'
import { Drawer, FloatButton, Button, Space } from 'antd';
import { useState } from 'react';
import { ToolOutlined, PushpinOutlined, FormOutlined, LinkOutlined, DownloadOutlined } from '@ant-design/icons';
import './ToolBar.css'

const ToolBar = () => {
  const [open, setOpen] = useState(false);
  const showDrawer = () => {
    open === false ? setOpen(true) : setOpen(false);
  };
  const onClose = () => {
    setOpen(false);
  };

  return (
    <>
      <FloatButton
        shape="square"
        style={{ right: 24, marginBottom: 20 }}
        icon={<ToolOutlined />}
        onClick={showDrawer}
      />
      <div className="tool-area">
        <Drawer 
          height={100}
          mask={false}
          placement="bottom" 
          closable={false}
          onClose={onClose} 
          open={open}
          getContainer={false}
        > 
          <div className="tool-button">
            <Space>
              <Button type="text" size="large" icon={<PushpinOutlined />} />
              <Button type="text" size="large" icon={<FormOutlined />} />
              <Button type="text" size="large" icon={<LinkOutlined />} />
              <Button type="text" size="large" icon={<DownloadOutlined />} />
              <Button type="text" size="large" icon={<img src="https://cloud.maptiler.com/static/img/maps/basic-v2.png?t=1663665773" width="40" height="40" alt='basic'></img>} />
              <Button type="text" size="large" icon={<img src="https://cloud.maptiler.com/static/img/maps/streets-v2.png?t=1663665773" width="40" height="40" alt='basic'></img>} />
              <Button type="text" size="large" icon={<img src="https://cloud.maptiler.com/static/img/maps/openstreetmap.png?t=1663665773" width="40" height="40" alt='basic'></img>} />
              <Button type="text" size="large" icon={<img src="https://cloud.maptiler.com/static/img/maps/hybrid.png?t=1663665773" width="40" height="40" alt='basic'></img>} />
            </Space>
          </div>
        </Drawer>
      </div>
      
    </>
  );
}

export default ToolBar