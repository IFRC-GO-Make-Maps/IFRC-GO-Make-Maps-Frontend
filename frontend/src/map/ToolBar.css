.tool-area {
  position: fixed;
  left: 50%;
  margin-left: -350px;
  bottom: 30px;
  width: 700px;
  height: 100px;
  text-align: center;
}

.ant-drawer .ant-drawer-body {
  padding: 15px;
}

.tool-button {
  padding: 10px;
}

.tool-area .ant-btn.ant-btn-lg.ant-btn-icon-only {
  width: 50px;
  padding-inline-start: 0;
  padding-inline-end: 0;
}

.tool-area .ant-btn.ant-btn-lg {
  font-size: 25px;
  height: 50px;
  padding: 6.428571428571429px 15px;
  border-radius: 8px;
}

.tool-area .ant-drawer-bottom>.ant-drawer-content-wrapper {
  bottom: 0;
  inset-inline: 0;
  box-shadow: 0 -6px 16px 0 rgb(0 0 0 / 8%), 0 -3px 6px -4px rgb(0 0 0 / 12%), 0 -9px 28px 8px rgb(0 0 0 / 5%);
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
}

.tool-area .ant-drawer .ant-drawer-content {
  width: 100%;
  height: 100%;
  overflow: auto;
  background: #ffffff;
  pointer-events: auto;
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
}