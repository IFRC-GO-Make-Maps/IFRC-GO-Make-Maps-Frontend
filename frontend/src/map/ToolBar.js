import React, { useState } from "react";
import { Drawer, FloatButton, Button, Space, Tooltip, Divider } from "antd";
import {
  PushpinOutlined,
  LinkOutlined,
  DownloadOutlined,
  LineOutlined,
  ExportOutlined,
} from "@ant-design/icons";
import { CgToolbox } from "react-icons/cg";
import { BiShapePolygon, BiPaint, BiLayer } from "react-icons/bi";
import "./ToolBar.css";

const ToolBar = ({
  handlePinButton,
  showLayerModal,
  showLinkModal,
  handlePaintButton,
  handleLineButton,
  handlePolygonButton,
  handleDownloadButton,
  handleExportButton,
}) => {
  const [open, setOpen] = useState(true);
  const [visual, setVisual] = useState(true);

  const showDrawer = () => {
    open === false ? setOpen(true) : setOpen(false);
    visual === false
      ? setVisual(true)
      : setTimeout(() => {
          setVisual(false);
        }, 300);
  };
  const onClose = () => {
    setOpen(false);
    setVisual(false);
  };

  return (
    <>
      <FloatButton
        shape="square"
        style={{ right: 24, marginBottom: -10, zIndex: 999 }}
        icon={<CgToolbox />}
        tooltip={<div>Tool Bar</div>}
        onClick={showDrawer}
      />
      <div
        className="tool-area"
        style={{ display: visual === false ? "none" : "initial", zIndex: 999 }}
      >
        <Drawer
          autoFocus={false}
          height={100}
          mask={false}
          placement="bottom"
          closable={false}
          onClose={onClose}
          open={open}
          getContainer={false}
        >
          <div className="tool-button">
            <Space>
              <Tooltip placement="top" title={<span>Pin</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<PushpinOutlined onClick={handlePinButton} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Layer</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<BiLayer onClick={showLayerModal} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Draw</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<BiPaint onClick={handlePaintButton} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Line</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<LineOutlined onClick={handleLineButton} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Polygon</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<BiShapePolygon onClick={handlePolygonButton} />}
                />
              </Tooltip>
              <Divider type="vertical" style={{ height: "2em" }} />
              <Tooltip placement="top" title={<span>Link</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<LinkOutlined onClick={showLinkModal} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Download</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<DownloadOutlined onClick={handleDownloadButton} />}
                />
              </Tooltip>
              <Tooltip placement="top" title={<span>Export GeoJson</span>}>
                <Button
                  type="text"
                  size="large"
                  icon={<ExportOutlined onClick={handleExportButton} />}
                />
              </Tooltip>
            </Space>
          </div>
        </Drawer>
      </div>
    </>
  );
};

export default ToolBar;
