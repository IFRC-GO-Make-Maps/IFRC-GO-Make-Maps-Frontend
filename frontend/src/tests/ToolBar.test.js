import React from "react";
import { render } from "@testing-library/react";
import ToolBar from "../map/ToolBar";

describe(ToolBar, () => {
  it("should render correctly", () => {
    const { container } = render(<ToolBar />);
    expect(container).toMatchSnapshot();
  });
});
