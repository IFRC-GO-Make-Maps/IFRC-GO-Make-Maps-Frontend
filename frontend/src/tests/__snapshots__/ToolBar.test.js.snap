// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar should render correctly 1`] = `
<div>
  <button
    class="css-dev-only-do-not-override-ixblex ant-float-btn ant-float-btn-default ant-float-btn-square"
    style="right: 24px; margin-bottom: -10px; z-index: 999;"
    type="button"
  >
    <div
      class="ant-float-btn-body"
    >
      <div
        class="ant-float-btn-content"
      >
        <div
          class="ant-float-btn-icon"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M17 5.5H20C21.1046 5.5 22 6.39543 22 7.5V19.5C22 20.6046 21.1046 21.5 20 21.5H4C2.89543 21.5 2 20.6046 2 19.5V7.5C2 6.39543 2.89543 5.5 4 5.5H7C7 3.84315 8.34315 2.5 10 2.5H14C15.6569 2.5 17 3.84315 17 5.5ZM14 4.5H10C9.44772 4.5 9 4.94772 9 5.5H15C15 4.94772 14.5523 4.5 14 4.5ZM20 7.5H4V9.5H20V7.5ZM4 19.5V11.5H7V13.5H11V11.5H13V13.5H17V11.5H20V19.5H4Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </button>
  <div
    class="tool-area"
    style="display: initial; z-index: 999;"
  >
    <div
      class="ant-drawer ant-drawer-bottom no-mask css-dev-only-do-not-override-ixblex ant-drawer-open ant-drawer-inline"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper"
        style="height: 100px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-body"
            >
              <div
                class="tool-button"
              >
                <div
                  class="ant-space css-dev-only-do-not-override-ixblex ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <span
                        aria-label="pushpin"
                        class="anticon anticon-pushpin"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="pushpin"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3a33.23 33.23 0 00-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 00-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8zM666.2 549.3l-24.5 24.5 3.8 34.4a259.92 259.92 0 01-30.4 153.9L262 408.8c12.9-7.1 26.3-13.1 40.3-17.9 27.2-9.4 55.7-14.1 84.7-14.1 9.6 0 19.3.5 28.9 1.6l34.4 3.8 24.5-24.5L608.5 224 800 415.5 666.2 549.3z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.061 22c1.523 0 2.84-.543 3.91-1.613 1.123-1.123 1.707-2.854 1.551-4.494l8.564-8.564a3.123 3.123 0 0 0-.002-4.414c-1.178-1.18-3.234-1.18-4.412 0l-8.884 8.884c-1.913.169-3.807 1.521-3.807 3.919 0 .303.021.588.042.86.08 1.031.109 1.418-1.471 2.208a1.001 1.001 0 0 0-.122 1.717C2.52 20.563 4.623 22 7.061 22c-.001 0-.001 0 0 0zM18.086 4.328a1.144 1.144 0 0 1 1.586.002 1.12 1.12 0 0 1 0 1.584L12 13.586 10.414 12l7.672-7.672zM6.018 16.423c-.018-.224-.037-.458-.037-.706 0-1.545 1.445-1.953 2.21-1.953.356 0 .699.073.964.206.945.475 1.26 1.293 1.357 1.896.177 1.09-.217 2.368-.956 3.107C8.865 19.664 8.049 20 7.061 20H7.06c-.75 0-1.479-.196-2.074-.427 1.082-.973 1.121-1.989 1.032-3.15z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <span
                        aria-label="line"
                        class="anticon anticon-line"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="line"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M904 476H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M20 14.185v-2.369A2.997 2.997 0 0 0 22 9c0-1.654-1.346-3-3-3a2.99 2.99 0 0 0-2.116.876L12.969 5.31c.01-.103.031-.204.031-.31 0-1.654-1.346-3-3-3S7 3.346 7 5c0 .762.295 1.451.765 1.981L6.091 9.212A2.977 2.977 0 0 0 5 9c-1.654 0-3 1.346-3 3s1.346 3 3 3c.159 0 .313-.023.465-.047L7.4 17.532c-.248.436-.4.932-.4 1.468 0 1.654 1.346 3 3 3a2.994 2.994 0 0 0 2.863-2.153l3.962-.792A2.987 2.987 0 0 0 19 20c1.654 0 3-1.346 3-3a2.995 2.995 0 0 0-2-2.815zM19 8a1.001 1.001 0 1 1-1 1c0-.551.448-1 1-1zm-9-4a1.001 1.001 0 1 1-1 1c0-.551.448-1 1-1zm-6 8a1.001 1.001 0 1 1 1 1c-.552 0-1-.449-1-1zm6 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm2.761-2.172A3.005 3.005 0 0 0 10 16c-.386 0-.752.079-1.091.213l-1.674-2.231C7.705 13.451 8 12.762 8 12c0-.536-.152-1.032-.399-1.467l1.935-2.58c.152.024.305.047.464.047a2.99 2.99 0 0 0 2.116-.876l3.915 1.566c-.01.103-.031.204-.031.31 0 1.302.839 2.401 2 2.815v2.369a2.996 2.996 0 0 0-2 2.815c0 .061.015.117.018.177l-3.257.652zM19 18a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M22 7.999a1 1 0 0 0-.516-.874l-9.022-5a1.003 1.003 0 0 0-.968 0l-8.978 4.96a1 1 0 0 0-.003 1.748l9.022 5.04a.995.995 0 0 0 .973.001l8.978-5A1 1 0 0 0 22 7.999zm-9.977 3.855L5.06 7.965l6.917-3.822 6.964 3.859-6.918 3.852z"
                        />
                        <path
                          d="M20.515 11.126 12 15.856l-8.515-4.73-.971 1.748 9 5a1 1 0 0 0 .971 0l9-5-.97-1.748z"
                        />
                        <path
                          d="M20.515 15.126 12 19.856l-8.515-4.73-.971 1.748 9 5a1 1 0 0 0 .971 0l9-5-.97-1.748z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <div
                      class="ant-divider css-dev-only-do-not-override-ixblex ant-divider-vertical"
                      role="separator"
                      style="height: 2em;"
                    />
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <span
                        aria-label="link"
                        class="anticon anticon-link"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="link"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M574 665.4a8.03 8.03 0 00-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 00-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 000 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 000 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 00-11.3 0L372.3 598.7a8.03 8.03 0 000 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <span
                        aria-label="download"
                        class="anticon anticon-download"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="download"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-text ant-btn-lg ant-btn-icon-only"
                      type="button"
                    >
                      <span
                        aria-label="export"
                        class="anticon anticon-export"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="export"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M888.3 757.4h-53.8c-4.2 0-7.7 3.5-7.7 7.7v61.8H197.1V197.1h629.8v61.8c0 4.2 3.5 7.7 7.7 7.7h53.8c4.2 0 7.7-3.4 7.7-7.7V158.7c0-17-13.7-30.7-30.7-30.7H158.7c-17 0-30.7 13.7-30.7 30.7v706.6c0 17 13.7 30.7 30.7 30.7h706.6c17 0 30.7-13.7 30.7-30.7V765.1c0-4.3-3.5-7.7-7.7-7.7zm18.6-251.7L765 393.7c-5.3-4.2-13-.4-13 6.3v76H438c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
